v1/src/test/java/com/google/cloud/teleport/bigtable/AvroToBigtableLT.java

```diff
@@ -0,0 +1,159 @@
+import com.google.cloud.teleport.it.bigtable.DefaultBigtableResourceManager;
+import com.google.cloud.teleport.it.launcher.PipelineLauncher.LaunchConfig;
+import com.google.cloud.teleport.it.launcher.PipelineLauncher.LaunchInfo;
+import com.google.cloud.teleport.it.launcher.PipelineOperator.Result;
+import com.google.cloud.teleport.metadata.TemplateLoadTest;
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+import com.google.common.io.Resources;
+import java.io.IOException;
+import java.text.ParseException;
+import java.time.Duration;
+import java.util.function.Function;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Performance tests for {@link AvroToBigtable GCS Avro to Bigtable} template. */
+@TemplateLoadTest(AvroToBigtable.class)
+@RunWith(JUnit4.class)
+public class AvroToBigtableLT extends TemplateLoadTestBase {
+  private static final String SPEC_PATH =
+      MoreObjects.firstNonNull(
+          TestProperties.specPath(), "gs://dataflow-templates/latest/GCS_Avro_to_Cloud_Bigtable");
+  private static final String ARTIFACT_BUCKET = TestProperties.artifactBucket();
+  private static final String TEST_ROOT_DIR = AvroToBigtableLT.class.getSimpleName().toLowerCase();
+  // 56,000,000 messages of the given schema approximately make up 10GB
+  private static final String NUM_MESSAGES = "56000000";
+  private static final String INPUT_PCOLLECTION = "Read from Avro/Read.out0";
+  private static final String OUTPUT_PCOLLECTION = "Transform to Bigtable.out0";
+  private static BigtableResourceManager bigtableResourceManager;
+  private static ArtifactClient artifactClient;
+  private static String generatorSchemaPath;
+  private static String avroSchemaPath;
+
+  @Before
+  public void setup() throws IOException {
+    // Set up resource managers
+    bigtableResourceManager =
+        DefaultBigtableResourceManager.builder(testName.getMethodName(), PROJECT).build();
+    Storage gcsClient = createGcsClient(CREDENTIALS);
+    artifactClient = GcsArtifactClient.builder(gcsClient, ARTIFACT_BUCKET, TEST_ROOT_DIR).build();
+    // upload schema files and save path
+    generatorSchemaPath =
+        getFullGcsPath(
+            ARTIFACT_BUCKET,
+            artifactClient
+                .uploadArtifact(
+                    "input/schema.json",
+                    Resources.getResource("AvroToBigtablePerformanceIT/schema.json").getPath())
+                .name());
+    avroSchemaPath =
+        getFullGcsPath(
+            ARTIFACT_BUCKET,
+            artifactClient
+                .uploadArtifact(
+                    "input/bigtable.avsc",
+                    Resources.getResource("schema/avro/bigtable.avsc").getPath())
+                .name());
+  }
+
+  @After
+  public void teardown() {
+    bigtableResourceManager.cleanupAll();
```

Comment:
nit:
I found that using `if (bigtableResourceManager != null)` around cleanups are a good practice for those tests.

If the setup/ResourceManager fails for any reason, this will be invoked and can throw a NullPointerException, so you end up getting some extra errors/red herrings on the output.

Reply:
Added checks in all performance tests